rules/builds.smk

```diff
@@ -141,43 +141,81 @@ rule mask:
+    if "priorities" in subsampling_settings and subsampling_settings["priorities"]["type"] == "proximity":
+        return f"results/{wildcards.build_name}/proximity_{subsampling_settings['priorities']['focus']}.tsv"
     else:
-        return config["subsample_focus"]["seq_per_group_regional"]
+        # TODO: find a way to make the list of input files depend on config
+        return config["files"]["include"]
 
-def _get_focus_exclude_argument_by_wildcards(wildcards):
-    if wildcards.region == "global":
-        return ""
+
+def get_priority_argument(wildcards):
+    subsampling_settings = _get_subsampling_settings(wildcards)
+
+    if "priorities" in subsampling_settings and subsampling_settings["priorities"]["type"] == "proximity":
+        return "--priority " + get_priorities(wildcards)
     else:
-        return f"--exclude-where \"region!={wildcards.region.replace('-', ' ')}\""
+        return ""
+
+
+def _get_specific_subsampling_setting(setting, optional=False):
+    def _get_setting(wildcards):
+        subsampling_scheme = _get_subsampling_scheme_by_build_name(wildcards.build_name)
+        geographic_name = _get_geographic_name_by_build_name(wildcards.build_name)
+        geographic_scale = _get_geographic_scale_by_build_name(wildcards.build_name)
+
+        if optional:
+            value = _get_subsampling_settings(wildcards).get(setting, "")
+        else:
+            value = _get_subsampling_settings(wildcards)[setting]
+
+        if isinstance(value, str):
+            value = value.format(**wildcards)
+        else:
+            return value
+
+        if ("geo_hierarchy" in config
+            and geographic_scale in config["geo_hierarchy"]
+            and geographic_name in config["geo_hierarchy"][geographic_scale]):
+            return value.format(**config["geo_hierarchy"][geographic_scale][geographic_name])
+        else:
+            return value
 
-rule subsample_focus:
+    return _get_setting
+
+rule subsample:
     message:
         """
-        Subsample all sequences into a focal set for {wildcards.region} with {params.sequences_per_group} per region
+        Subsample all sequences into a {wildcards.subsample} set for build '{wildcards.build_name}' with {params.sequences_per_group} per {params.group_by}
         """
     input:
         sequences = rules.mask.output.alignment,
         metadata = rules.download.output.metadata,
-        include = config["files"]["include"]
+        include = config["files"]["include"],
+        priorities = get_priorities
     output:
-        sequences = REGION_PATH + "subsample_focus.fasta"
-    log:
-        "logs/subsample_focus_{region}.txt"
+        sequences = "results/{build_name}/sample-{subsample}.fasta"
     params:
-        group_by = _get_group_by_wildcards,
-        sequences_per_group = _get_sequences_per_group_by_wildcards,
-        exclude_argument = _get_focus_exclude_argument_by_wildcards
+        group_by = _get_specific_subsampling_setting("group_by"),
```

Comment:
Perhaps for another PR, but can we reformulate `_get_specific_subsampling_setting` to return "--group_by X Y Z" so that we can allow a subsampling (sub)scheme which doesn't define a "group_by" (in this case the empty string would be returned and)